src/components/ThemeProvider/README.md

```diff
@@ -0,0 +1,232 @@
+    colorScheme: 'dark',
+  }}
+>
+  <Card
+    title="Shipment 1234"
+    secondaryFooterActions={[{content: 'Edit shipment'}]}
+    primaryFooterAction={{content: 'Add tracking number'}}
+  >
+    <Card.Section title="Items">
+      <List>
+        <List.Item>1 × Oasis Glass, 4-Pack</List.Item>
+        <List.Item>1 × Anubis Cup, 2-Pack</List.Item>
+      </List>
+    </Card.Section>
+  </Card>
+</AppProvider>
+```
+
+### Theme provider with a color scheme nested within an app provider
+
+A theme provider can be nested within the theme provider rendered by the app provider in order to override the color scheme at a local level.
+
+```jsx
+<AppProvider i18n={{}} features={{newDesignLanguage: true}}>
+  <TextContainer>
+    <Card
+      title="Shipment 1234"
+      secondaryFooterActions={[{content: 'Edit shipment'}]}
+      primaryFooterAction={{content: 'Add tracking number'}}
+    >
+      <Card.Section title="Items">
+        <List>
+          <List.Item>1 × Oasis Glass, 4-Pack</List.Item>
+          <List.Item>1 × Anubis Cup, 2-Pack</List.Item>
+        </List>
+      </Card.Section>
+    </Card>
+    <ThemeProvider theme={{colorScheme: 'dark'}}>
+      <Card
+        title="Shipment 1234"
+        secondaryFooterActions={[{content: 'Edit shipment'}]}
+        primaryFooterAction={{content: 'Add tracking number'}}
+      >
+        <Card.Section title="Items">
+          <List>
+            <List.Item>1 × Oasis Glass, 4-Pack</List.Item>
+            <List.Item>1 × Anubis Cup, 2-Pack</List.Item>
+          </List>
+        </Card.Section>
+      </Card>
+    </ThemeProvider>
+  </TextContainer>
+</AppProvider>
+```
+
+### Theme provider with colors nested within an app provider
+
+A theme provider can be nested within the theme provider rendered by the app provider in order to override colors at a local level.
+
+```jsx
+<AppProvider i18n={{}} features={{newDesignLanguage: true}}>
+  <Card>
+    <Card.Section>
+      <Stack alignment="center">
+        <Stack.Item fill>
+          This setting is <TextStyle variation="strong">disabled</TextStyle>.
+        </Stack.Item>
+        <Stack.Item>
+          <Button primary>Enable</Button>
+        </Stack.Item>
+      </Stack>
+    </Card.Section>
+    <Card.Section>
+      <Stack alignment="center">
+        <Stack.Item fill>
+          This setting is <TextStyle variation="strong">disabled</TextStyle>.
+        </Stack.Item>
+        <Stack.Item>
+          <ThemeProvider theme={{colors: {primary: '#3b5998'}}}>
+            <Button primary>Enable</Button>
+          </ThemeProvider>
+        </Stack.Item>
+      </Stack>
+    </Card.Section>
+  </Card>
+</AppProvider>
+```
+
+---
+
+## Consuming theme colors in a component
+
+The theme provider component uses [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) to share color values with components. For a full list of available CSS custom properties, see the [Polaris tokens docs](link/to/polaris/tokens/docs).
```

Comment:
Just noticed this link is broken which causes the Style Guide build to fail (link checker)—might require a patch release. For now this can point to https://github.com/Shopify/polaris-react/blob/master/documentation/Color%20system.md.